### PR TITLE
SlotFill: fix dependencies of registration effects, deduplicate code

### PR DESCRIPTION
**Pull request by jsnajdr**
This PR tidies up the code for a `Fill` registration and rendering.

For registration and unregistration, there used to be two effects: one runs on mount/unmount, another on `name` change. They can be easily merged into one with a `[ name ]` dependency. Registration and unregistration still happens at exactly the right times. And also the `ref.current.name` value is updated when `name` changes.

There's also a second effect that updates `ref.current.children` and rerenders the slot when `children` change. This one needs additional `slot` dependency. The `slot` object is constant the entire time a slot is mounted.

### CODE REVIEW

File: packages/components/src/slot-fill/fill.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import useSlot from './use-slot';
 import type { FillComponentProps } from './types';
 
 export default function Fill( { name, children }: FillComponentProps ) {
-	const { registerFill, unregisterFill } = useContext( SlotFillContext );
+	const registry = useContext( SlotFillContext );
```
Comment (youknowriad) [inline packages/components/src/slot-fill/fill.ts:14]: Is this a stable value?

**Review by youknowriad — APPROVED** [review]
This simplifies things a lot. It's way more clear now.

File: packages/components/src/slot-fill/fill.ts (hunk 1 above)
Comment (jsnajdr) [reply packages/components/src/slot-fill/fill.ts:14]: Yes, as long as the context provider is the same, the `registry` is a constant object.

**Review by Mamaduka — APPROVED** [review]
Nice improvement. Thank you, @jsnajdr!

**Comment by jsnajdr** [comment]
For some reason the e2es are failing repeatedly. It's a low level failure -- failed WP database connection.

**Pull request merged**